git-remote-ipfs/src/Network/IPFS/Git/RemoteHelper/Trans.hs

```diff
@@ -188,6 +192,7 @@ newEnv envLogger envOptions = do
     envVerbosity <- newIORef 1
     envDryRun    <- newIORef False
     envGit       <- findRepo >>= openRepo
+    envLobs      <- newMVar Nothing
```

Comment:
Do we have a good reason for having a `Maybe` here and not, say, initialise the MVar empty by calling `newEmptyMVar` so that the first reader has to block up until that MVar is not filled? Is this to avoid too much contention perhaps?

Reply:
`modifyMVar` will block until the `MVar` becomes full, so I would need some other way to signal that I want it filled now, and also make sure only one thread is attempting to fill it. Using `MVar (Maybe a)` seemed like an easy way to achieve this (essentially, poor man's memoisation).

Reply:
I see, 👌 .